plugin/pkg/scheduler/scheduler.go

```diff
@@ -186,6 +190,13 @@ func (s *Scheduler) scheduleOne() {
 		return
 	}
 
+	// Optimistically assume that the binding will succeed, so we need to invalidate affected
```

Comment:
can we add a unit test for this optimisitc check logic since its a default behaviour we want to document?